_rules/sequentially-focusable-element-has-visible-focus-oj04fd.md

```diff
@@ -0,0 +1,181 @@
+---
+id: oj04fd
+name: Element in sequential focus order has visible focus
+rule_type: atomic
+description: |
+  This rule checks that each element in sequential focus order has some visible focus indication.
+accessibility_requirements:
+  wcag20:2.4.7: # Focus Visible (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - CSS styling
+  - DOM tree
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+---
+
+## Applicability
+
+The rule applies to any element which is part of [sequential focus navigation][] in an [HTML document](https://html.spec.whatwg.org/#document) with at least two [focusable][] elements.
+
+## Expectation
+
+For each target element, there is at least one device pixel inside the [scrolling area][] of the [viewport][] whose color is different when the element is [focused][] from when it is not.
+
+## Assumptions
+
+This rule assumes that documents with only one focusable element automatically satisfy [Success Criterion 2.4.7 Focus Visible][sc247], as stated in [Understanding Success Criterion 2.4.7: Focus Visible][usc247]: "if there is only one keyboard actionable control on the screen, the success criterion would be met".
```

Comment:
The keyword here seems to be "keyboard actionable". It therefore seems like there's a disconnect between this assumption and the applicability of the rule as the latter requires at least two "focusable" elements in the document, not at least two "keyboard actionable" elements.

Reply:
And given that @Jym77 has already created the definition for [keyboard actionable](https://github.com/act-rules/act-rules.github.io/blob/bf57d85c3f0514721dc70868506b3519a2e65bda/pages/glossary/keyboard-actionable-element.md) that should be an easy fix 🙂

Reply:
Interesting discussion…
I am not convinced that our "keyboard actionable" make sense here. If an element can be tabbed but not clicked, it still require focus indication. Typically, [HTML specs](https://html.spec.whatwg.org/multipage/interaction.html#data-model:sequentially-focusable) have an example where a UA could make bullets (the bullet symbol) in a list could be sequentially focusable as a reading guide, in which case they definitely need an indicator…

OTOH, WCAG (the SC) is only about "a keyboard operable UI component"… Not sure whether these bullets would be considered as a UI component by WCAG… But "operable" could be understand as "tabbable", if the only purpose is to be tabbable (reading guide, …)

Suggestion: put the applicability as "if there are at least 2 sequentially focusable elements" (rather than "at least 2 focusable)"; and add an assumption "this rule assumes that sequentially focusable elements are UI components", which sounds safe.
+ update all examples since the dummy button with `tabindex=-1` does not work any more 😕 

This would also avoid the need for "keyboard actionable" which is annoying to test automatically. So the Applicability could be automated, and the rule also in many cases (e.g., detecting an indicator through styling, …)

@carlosapaduarte @kasperisager WDYT?

Reply:
Something along the lines of "if the sequential focus navigation order contains more than 1 element" works fine by me 👍

Reply:
Agreed